### PR TITLE
Sample consensus: solve maybe-uninitialized warning in line model

### PR DESCRIPTION
**Pull request by mvieth**
warning was: ‘centroid’ may be used uninitialized in this function [-Wmaybe-uninitialized]

### CODE REVIEW

File: sample_consensus/include/pcl/sample_consensus/impl/sac_model_line.hpp [hunk 1]
```diff
@@ -221,7 +221,12 @@ pcl::SampleConsensusModelLine<PointT>::optimizeModelCoefficients (
 
   // Compute the 3x3 covariance matrix
   Eigen::Vector4f centroid;
-  compute3DCentroid (*input_, inliers, centroid);
+  if (0 == compute3DCentroid (*input_, inliers, centroid))
+  {
+    PCL_DEBUG ("[pcl::SampleConsensusModelLine::optimizeModelCoefficients] compute3DCentroid failed (returned 0), possibly because there are no valid inliers.\n");
```
Comment (larshg) [inline sample_consensus/include/pcl/sample_consensus/impl/sac_model_line.hpp:226]: The amount of inliers is checked to be minimum sample_size_ in line 213? But if this is 0..?

Comment (mvieth) [reply sample_consensus/include/pcl/sample_consensus/impl/sac_model_line.hpp:226]: The return value can be smaller than `inliers.size()` if not all inliers are finite

Comment (larshg) [reply sample_consensus/include/pcl/sample_consensus/impl/sac_model_line.hpp:226]: Ahh, true.

**Review by larshg — APPROVED** [review]

File: sample_consensus/include/pcl/sample_consensus/impl/sac_model_line.hpp (hunk 1 above)
Comment (koide3) [inline sample_consensus/include/pcl/sample_consensus/impl/sac_model_line.hpp:226]: Should use PCL_WARN?

Comment (mvieth) [reply sample_consensus/include/pcl/sample_consensus/impl/sac_model_line.hpp]: Yes, that is more appropriate.

**Review by larshg — APPROVED** [review]

File: sample_consensus/include/pcl/sample_consensus/impl/sac_model_line.hpp [hunk 2]
```diff
@@ -221,7 +221,12 @@ pcl::SampleConsensusModelLine<PointT>::optimizeModelCoefficients (
 
   // Compute the 3x3 covariance matrix
   Eigen::Vector4f centroid;
-  compute3DCentroid (*input_, inliers, centroid);
+  if (0 == compute3DCentroid (*input_, inliers, centroid))
```
Comment (kunaltyagi) [inline sample_consensus/include/pcl/sample_consensus/impl/sac_model_line.hpp:224]: Any particular reason for using `0 == compute()`? A simple `if (!compute())` will also work

File: sample_consensus/include/pcl/sample_consensus/impl/sac_model_line.hpp [hunk 3]
```diff
@@ -221,7 +221,12 @@ pcl::SampleConsensusModelLine<PointT>::optimizeModelCoefficients (
...
+  {
+    PCL_WARN ("[pcl::SampleConsensusModelLine::optimizeModelCoefficients] compute3DCentroid failed (returned 0), possibly because there are no valid inliers.\n");
```
Comment (kunaltyagi) [inline sample_consensus/include/pcl/sample_consensus/impl/sac_model_line.hpp:226]: Might make sense to reword this to eliminate `possibly`. The only reason for a 0 return is if `inliers.empty()` or `all(points == NaN)`

File: sample_consensus/include/pcl/sample_consensus/impl/sac_model_line.hpp (hunk 2 above)
Comment (larshg) [reply sample_consensus/include/pcl/sample_consensus/impl/sac_model_line.hpp:224]: I think compute3DCentroid counts the number of points used, so comparing with 0 gives a clearer understanding of what is returned - you suggestion makes an implicit conversion to bool?

File: sample_consensus/include/pcl/sample_consensus/impl/sac_model_line.hpp (hunk 3 above)
Comment (mvieth) [reply sample_consensus/include/pcl/sample_consensus/impl/sac_model_line.hpp]: Makes sense, I removed "possibly"

**Review by kunaltyagi — APPROVED** [review]
